Tools/AP_Periph/can.cpp

```diff
@@ -2356,6 +2359,192 @@ void AP_Periph_FW::can_send_ADSB(struct __mavlink_adsb_vehicle_t &msg)
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_DETONATION_OBSERVED;
+            break;
+        }
+
+        switch (state.misfire_status) {
+        case Misfire_Status::NOT_SUPPORTED:
+            break;
+        case Misfire_Status::NOT_OBSERVED:
+            pkt.flags |=
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_MISFIRE_SUPPORTED;
+            break;
+        case Misfire_Status::OBSERVED:
+            pkt.flags |=
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_MISFIRE_SUPPORTED |
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_MISFIRE_OBSERVED;
+            break;
+        }
+
+        switch (state.debris_status) {
+        case Debris_Status::NOT_SUPPORTED:
+            break;
+        case Debris_Status::NOT_DETECTED:
+            pkt.flags |=
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_DEBRIS_SUPPORTED;
+            break;
+        case Debris_Status::DETECTED:
+            pkt.flags |=
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_DEBRIS_SUPPORTED |
+                UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_FLAG_DEBRIS_DETECTED;
+            break;
+        }
+
+        pkt.engine_load_percent = state.engine_load_percent;
+        pkt.engine_speed_rpm = state.engine_speed_rpm;
+        pkt.spark_dwell_time_ms = state.spark_dwell_time_ms;
+        pkt.atmospheric_pressure_kpa = state.atmospheric_pressure_kpa;
+        pkt.intake_manifold_pressure_kpa = state.intake_manifold_pressure_kpa;
+        pkt.intake_manifold_temperature = state.intake_manifold_temperature;
+        pkt.coolant_temperature = state.coolant_temperature;
+        pkt.oil_pressure = state.oil_pressure;
+        pkt.oil_temperature = state.oil_temperature;
+        pkt.fuel_pressure = state.fuel_pressure;
+        pkt.fuel_consumption_rate_cm3pm = state.fuel_consumption_rate_cm3pm;
+        pkt.estimated_consumed_fuel_volume_cm3 = state.estimated_consumed_fuel_volume_cm3;
+        pkt.throttle_position_percent = state.throttle_position_percent;
+        pkt.ecu_index = state.ecu_index;
+        pkt.spark_plug_usage = uint8_t(state.spark_plug_usage);
+
+        // assume single set of cylinder status
+        pkt.cylinder_status.len = 1;
+        auto &c = pkt.cylinder_status.data[0];
+        c.ignition_timing_deg = state.cylinder_status[0].ignition_timing_deg;
+        c.injection_time_ms = state.cylinder_status[0].injection_time_ms;
+        c.cylinder_head_temperature = state.cylinder_status[0].cylinder_head_temperature;
+        c.exhaust_gas_temperature = state.cylinder_status[0].exhaust_gas_temperature;
+        c.lambda_coefficient = state.cylinder_status[0].lambda_coefficient;
+
+        uint8_t buffer[UAVCAN_EQUIPMENT_ICE_RECIPROCATING_STATUS_MAX_SIZE] {};
+        uint16_t total_size = uavcan_equipment_ice_reciprocating_Status_encode(&pkt, buffer, !periph.canfdout());
```

Comment:
`const`.  Had to be said.


No it didn't.

Reply:
added